modules/installation-azure-permissions.adoc

```diff
@@ -9,7 +9,7 @@
 {product-title} needs a service principal so it can manage Microsoft Azure resources. Before you can create a service principal, your Azure account subscription must have the following roles:
 
 * `User Access Administrator`
-* `Owner`
+* `Contributor`
```

Comment:
Here lists the roles for azure account to create service principal, but not role for service principal which is used by openshift installer. Do you think that it is necessary to update `Owner` to `Contributor`? @chiragkyal

Reply:
Got that. However, AFAIK `Owner` is the highest access, so if we want to keep `Owner` here, then `User Access Administrator` may not be required. 

And I think if an azure account has `Contributor` + `User Access Administrator` roles, then also it can create a SP and assign those permissions to it. @jinyunma WDYT ?

Reply:
As per [azure doc](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#contributor)

**Owner**
```json
{
    "permissions": [
      {
        "actions": [
          "*"
        ],
        "notActions": []
      }
    ],
    "roleName": "Owner",
    "roleType": "BuiltInRole"
  }
```


**Contributor** 

```json
{
  "permissions": [
    {
      "actions": [
        "*"
      ],
      "notActions": [
        "Microsoft.Authorization/*/Delete",
        "Microsoft.Authorization/*/Write",
        "Microsoft.Authorization/elevateAccess/Action",
        "Microsoft.Blueprint/blueprintAssignments/write",
        "Microsoft.Blueprint/blueprintAssignments/delete",
        "Microsoft.Compute/galleries/share/action"
      ],
    }
  ],
  "roleName": "Contributor",
  "roleType": "BuiltInRole",
}
```


**User Access Administrator**

```json
{
  "permissions": [
    {
      "actions": [
        "*/read",
        "Microsoft.Authorization/*",
        "Microsoft.Support/*"
      ],
      "notActions": []
    }
  ],
  "roleName": "User Access Administrator",
  "roleType": "BuiltInRole",
}
```

----
So what I've understood, if someone has `Contributor` + `User Access Administrator` access, it's near to `Owner` access, except the following permissions, which are added as notActions in `Contributor`.  

```
- Microsoft.Blueprint/blueprintAssignments/write	
- Microsoft.Blueprint/blueprintAssignments/delete	
- Microsoft.Compute/galleries/share/action
```

So, if  these 3 permissions are not required at any point in time, then Contributor + User Access Administrator should work fine, I believe. @jinyunma thoughts ?

Reply:
@chiragkyal thanks for so detailed explanation. Based on your comments, I guess it should work fine for role Contributor + User Access Administrator. But still need to verify that. 
Currently, my azure user account have role "Owner" on subscription level, and I have no right to update role.
File a ticket https://issues.redhat.com/browse/DPP-12337 to request role updating on my account.
Will verify it when role is updated.

Reply:
Hi @chiragkyal , my azure account has been updated and only have role "Contributor" and "User Access Administrator" on subscription "Openshift QE“，then it is successful to create sp following [installation doc](https://docs.openshift.com/container-platform/4.12/installing/installing_azure/installing-azure-account.html#installation-azure-service-principal_installing-azure-account) .
so it should be okay to change role from "Owner" to "Contributor".

Reply:
@jinyunma UPI part also needs to be updated.

Reply:
ah, right. @xenolinux please also help to update role in [UPI doc](https://docs.openshift.com/container-platform/4.12/installing/installing_azure/installing-azure-user-infra.html#installation-azure-permissions_installing-azure-user-infra).

Reply:
The module `installation-azure-permissions.adoc` is included in both UPI and IPI-related assembly files. So we don't need any additional changes for UPI in installation-azure-permissions.adoc file.